integration/vite-basename-dev-test.ts

```diff
@@ -0,0 +1,140 @@
+import fs from "node:fs/promises";
+import path from "node:path";
+import type { Page } from "@playwright/test";
+import { test, expect } from "@playwright/test";
+import getPort from "get-port";
+
+import { createProject, VITE_CONFIG, viteDev } from "./helpers/vite.js";
+
+const files = {
+  "app/routes/_index.tsx": String.raw`
+    import { useState, useEffect } from "react";
+    import { Link } from "@remix-run/react"
+
+    export default function IndexRoute() {
+      const [mounted, setMounted] = useState(false);
+      useEffect(() => {
+        setMounted(true);
+      }, []);
+
+      return (
+        <div id="index">
+          <h2 data-title>Index</h2>
+          <input />
+          <p data-mounted>Mounted: {mounted ? "yes" : "no"}</p>
+          <p data-hmr>HMR updated: 0</p>
+          <Link to="/other">other</Link>
+        </div>
+      );
+    }
+  `,
+  "app/routes/other.tsx": String.raw`
+    import { useLoaderData } from "@remix-run/react";
+
+    export const loader = () => {
+      return "other-loader";
+    };
+
+    export default function OtherRoute() {
+      const loaderData = useLoaderData()
+
+      return (
+        <div id="other">
+          <p>{loaderData}</p>
+        </div>
+      );
+    }
+  `,
+};
+
+test.describe(() => {
+  let port: number;
+  let cwd: string;
+  let stop: () => unknown;
+
+  test.beforeAll(async () => {
+    port = await getPort();
+    cwd = await createProject({
+      "vite.config.js": await VITE_CONFIG({
+        port,
+        viteOptions: '{ base: "/mybase/" }',
+        pluginOptions: '{ publicPath: "/mybase/" }',
+      }),
```

Comment:
I refactored this test to use the `VITE_CONFIG` file helper. To make this work I've also added a `viteOptions` to it.